app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -37,106 +36,102 @@ interface PropsWithoutHref {
-
-/**
- * The position of the icon in the button
- */
-export type IconPosition = 'end' | 'start'
-/**
- * The variant of the button
- */
-export type Variant =
-  | 'cancel'
-  | 'custom'
-  | 'delete'
-  | 'icon'
-  | 'link'
-  | 'outline'
-  | 'primary'
-  | 'submit'
-  | 'tertiary'
-
-const DEFAULT_CLASSES =
-  'flex whitespace-nowrap cursor-pointer border border-transparent transition-[opacity,outline-offset,background,border-color] duration-150 ease-in-out select-none text-center items-center justify-center'
-const FOCUS_CLASSES =
-  'focus-visible:outline-offset-2 focus:outline-none focus-visible:outline focus-visible:outline-primary'
-const EXTRA_CLICK_ZONE_CLASSES = 'flex relative after:inset-[-12px] after:absolute'
-const DISABLED_CLASSES = 'disabled:opacity-50 disabled:cursor-not-allowed'
-const LOADING_CLASSES = 'cursor-wait'
-const FULL_WIDTH_CLASSES = 'w-full'
-
-const CLASSES_FOR_SIZE: Record<Size, string> = {
-  hero: 'px-8 py-4 text-lg',
-  large: 'px-6 py-3 text-base',
-  medium: 'px-4 py-2 text-sm',
-  small: 'px-3 py-1 text-xs',
-  xsmall: 'px-2 py-1 text-xs',
-  xxsmall: 'px-1.5 py-0.5 text-xs',
-  custom: '',
-}
-
-const CLASSES_FOR_VARIANT: Record<Variant, string> = {
-  custom: '',
-  link: 'inline-flex px-0 py-0 rounded-sm text-primary/50 underline hover:text-primary',
-  primary: 'bg-primary text-white hover:bg-primary/70',
-  tertiary: 'bg-share text-white hover:bg-share/90',
-  cancel: 'bg-selected-frame opacity-80 hover:opacity-100',
-  delete: 'bg-delete text-white',
-  icon: 'opacity-70 hover:opacity-100',
-  submit: 'bg-invite text-white opacity-80 hover:opacity-100',
-  outline: 'border-primary/40 text-primary font-bold hover:border-primary/90',
-}
-
-const CLASSES_FOR_ROUNDING: Record<Roundings, string> = {
-  full: 'rounded-full',
-  large: 'rounded-lg',
-  medium: 'rounded-md',
-  none: 'rounded-none',
-  small: 'rounded-sm',
-  xlarge: 'rounded-xl',
+  readonly onPress?: (event: aria.PressEvent) => Promise<void> | void
 }
 
-const ICON_POSITION: Record<IconPosition, string> = {
-  start: '',
-  end: 'flex-row-reverse',
-}
+export const BUTTON_STYLES = twv.tv({
+  base: 'group flex whitespace-nowrap cursor-pointer border border-transparent transition-[opacity,outline-offset,background,border-color] duration-150 ease-in-out select-none text-center items-center justify-center appearance-none',
+  variants: {
+    isDisabled: { true: 'disabled:opacity-50 disabled:cursor-not-allowed' },
+    isFocused: {
+      true: 'focus:outline-none focus-visible:outline focus-visible:outline-primary',
+    },
+    loading: { true: { base: 'cursor-wait', content: 'opacity-0' } },
+    fullWidth: { true: 'w-full' },
+    size: {
+      custom: '',
+      hero: 'px-8 py-4 text-lg',
+      large: 'px-6 py-3 text-base',
+      medium: 'px-4 py-2 text-sm',
+      small: 'px-3 py-1 text-xs',
+      xsmall: 'px-2 py-1 text-xs',
+      xxsmall: 'px-1.5 py-0.5 text-xs',
+    },
+    iconOnly: { true: '' },
+    rounding: {
```

Comment:
i'd probably prefer `rounded='lg'` (for example) to match the tailwind classnames to avoid having to learn a naming convention that is *slightly* different, but similar enough that it'd end up confusing the user